### DISCUSSION TITLE
Strongswan send's down event's infinitely, is there any way to slow down these events?

### DISCUSSION THREAD

**Discussion by digambaringale333** [strongswan/strongswan#684] [Q&A]
Hi,
In our application we are using DAVICI API's to create IPSec tunnel, in some failure scenarios we have seen that VICI keep's sending down event's (time interval is in ms) continuously. Is there any way to slow down this events (by setting any parameter's in conn or strongswan.conf or is there any other way to increase the interval between two down event's)?

our child section start action is set to trap

**Comment by tobiasbrunner** [comment]
Please provide more information. What failure scenarios? What down events? What interval? Are you sure it's not just an error in your implementation?

**Reply by digambaringale333** [reply]
OK.. Let me add some more information here.
1. We have registered for IKE UP/DOWN CHILD UP/DOWN events with VICI.
2. When we are trying to initiate a tunnel where we dont get successful response from it's peer in error case (for eg.IKE SA stuck's in connecting state on responder site- it's fine because we are intentionally stimulating error scenarios to handle them gracefully)
3. in above case, we could see that VICI is streaming IKE DOWN event's contentiously (the time interval between two consecutive IKE DOWN event's is in millisecond's). Can we slow down these IKE down event's?

**Reply by tobiasbrunner** [reply]
There should be no down events at all in that case (i.e. if there never was an up event, which isn't the case for an IKE_SA in connecting state). And even if there was one, why should there be multiple for a single SA? This really sounds more like an implementation error on your part.

**Reply by digambaringale333** [reply]
In good path scenario we could see that everything work's fine, we get one IKE UP after initiate command and when we terminate tunnel we get one DOWN event (that's why I thought we are registering with IKE UP/DOWN event's properly ). We could see this issue(Continuous streaming of IKE DOWN events ) only when there is issue with tunnel creation. Do you think start action in child section play's any role here? We have trap policy in child section.

**Reply by digambaringale333** [reply]
Do you want to say that there will be never IKE down event's streamed continuously from VICI? There will be always 1-1 mapping between IKE UP and IKE down events  (If there is only one IKE UP then there will be only one corresponding IKE DOWN event)?

**Reply by tobiasbrunner** [reply]
Yes

**Reply by digambaringale333** [reply]
Thanks for your reply Tobias, your input always helps us to understand strongswan better.

**Reply by tobiasbrunner** [reply]
So were you able to fix the issue?

**Reply by digambaringale333** [reply]
We are debugging it from different perspective now (We are just dumping the IKE_UPDOWN event's in a file where I could see that there is always new unique id with each event it means it's generate by strongswan, we are not polling the socket multiple times).
Above mentioned issues(event's) are on responder side, but on the initiator side now we can see our application is sending multiple times initiate command (we are sending contineous initiate command's if we get failure for earlier one that must be causing the continuous IKE UP/DOWN event's on responder side) we might have to slow down the initiate operation from our application (If we get failure for initiate command then we have to wait for sometime and then may be we can retry initiate operation).

**Reply by tobiasbrunner** [reply]
> where I could see that there is always new unique id with each event

Unique ID of the IKE_SA?

> Above mentioned issues(event's) are on responder side

What do you mean? Didn't you say the test scenario was when the initiator can't reach the responder? So in what state are the SAs on the responder and why are they terminated?

> sending multiple times initiate command (we are sending contineous initiate command's if we get failure for earlier one that must be causing the continuous IKE UP/DOWN event's on responder side)

You don't wait at all? You just send initiate command after initiate command? And you mentioned that you use trap policies, so why send initiate at all? (Just generate traffic to trigger it.)

**Reply by digambaringale333** [reply]
> 
> 
> > where I could see that there is always new unique id with each event
> 
> Unique ID of the IKE_SA?
> Yes.
> > Above mentioned issues(event's) are on responder side
> 
> What do you mean? Didn't you say the test scenario was when the initiator can't reach the responder? So in what state are the SAs on the responder and why are they terminated?
> 
> > sending multiple times initiate command (we are sending continuous initiate command's if we get failure for earlier one that must be causing the continuous IKE UP/DOWN event's on responder side)
> 
> You don't wait at all? You just send initiate command after initiate command? And you mentioned that you use trap policies, so why send initiate at all? (Just generate traffic to trigger it.)

> Yes, when we get the failure from VICI for earlier initiate operation our then application immediately retries to issue initiate operation again. Our application is layered one, we get initiate operation request from upper layer(this layer is common for all of the other components), that's why it keep's retrying if gets failure. That's where we have to slow it down initiate retires.

**Reply by tobiasbrunner** [reply]
> Yes, when we get the failure from VICI for earlier initiate operation

But that takes some time, no? Or do you get such failures within milliseconds?
